Week2_LibraryTask/test/Library.ts

```diff
@@ -0,0 +1,165 @@
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+describe("Library", function () {
+  // We define a fixture to reuse the same setup in every test.
+  // We use loadFixture to run this setup once, snapshot that state,
+  // and reset Hardhat Network to that snapshot in every test.
+  async function deployLibrary() {
+
+    // Contracts are deployed using the first signer/account by default
+    const [owner, otherAccount] = await ethers.getSigners();
+
+    const LibraryFactory = await ethers.getContractFactory("Library");
+    const library = await LibraryFactory.deploy();
+
+    return { library, owner, otherAccount };
+  }
+
+  async function deployLibraryAndOneBook() {
+
+    // Contracts are deployed using the first signer/account by default
+    const [owner, otherAccount] = await ethers.getSigners();
+
+    const LibraryFactory = await ethers.getContractFactory("Library");
+    const library = await LibraryFactory.deploy();
+    const bookTitle = "Harry Potter I";
+    const bookCopies = 1;
+
+    await library.addBooks(bookTitle, bookCopies);
+
+    const keccak256 = require('keccak256')
+    const titleHash = keccak256(bookTitle);
+
+    return { library, owner, otherAccount, titleHash, bookTitle };
+  }
+
+  async function deployLibraryAndManyBooks() {
+    // Contracts are deployed using the first signer/account by default
+    const [owner, otherAccount] = await ethers.getSigners();
+
+    const LibraryFactory = await ethers.getContractFactory("Library");
+    const library = await LibraryFactory.deploy();
+    const bookTitle = "Harry Potter I";
+    const bookCopies = 3;
+
+    await library.addBooks(bookTitle, bookCopies);
+
+    const keccak256 = require('keccak256')
+    const titleHash = keccak256(bookTitle);
+
+    return { library, owner, otherAccount, titleHash, bookTitle };
+  }
+
+  describe("Deployment", function () {
+    
+    it("Should set the right owner", async function () {
+      const { library, owner } = await loadFixture(deployLibrary);
```

Comment:
Personally i prefer using beforeEach instead so we have leaner tests, however both beforeEach and loadFixture work

Reply:
I was debating on this since I am using different fixtures for the different tests, I have three different fixtures. Is it a better practice to use beforeEach and to do the specific test set-up inside each test?